aeron-driver/src/main/java/io/aeron/driver/DriverConductor.java

```diff
@@ -1790,15 +1868,74 @@ private NetworkPublication newNetworkPublication(
         }
     }
 
+    private  String enrichNetworkPublicationChannelUri(
+        final ChannelUri channelUri,
+        final int streamId,
+        final UdpChannel udpChannel,
+        final SendChannelEndpoint channelEndpoint,
+        final FlowControl flowControl,
+        final PublicationParams params)
+    {
+        // todo
+
+        final ChannelUriStringBuilder uriStringBuilder = new ChannelUriStringBuilder(channelUri);
+
+        configureUriStringBuilderWithPublicationParams(params, uriStringBuilder);
+
+
+        uriStringBuilder.streamId(streamId);
+
+
+        uriStringBuilder.flowControl(flowControl.getClass().getName());
+        final String ccStr = udpChannel.channelUri().get(CONGESTION_CONTROL_PARAM_NAME);
+        uriStringBuilder.congestionControl(null == ccStr ? "static" : ccStr);
+
+        uriStringBuilder.socketRcvbufLength(channelEndpoint.socketRcvbufLength());
+        uriStringBuilder.socketSndbufLength(channelEndpoint.socketSndbufLength());
+
+        if (udpChannel.isMulticast() && !udpChannel.hasMulticastTtl())
+        {
+            uriStringBuilder.ttl(ctx.socketMulticastTtl());
+        }
+
+//        int publicationWindowLength;
+
+        return uriStringBuilder.toString();
+    }
+
+    private static void configureUriStringBuilderWithPublicationParams(PublicationParams params, ChannelUriStringBuilder uriStringBuilder)
+    {
+        uriStringBuilder.linger(params.lingerTimeoutNs);
+
+        uriStringBuilder.untetheredWindowLimitTimeoutNs(params.untetheredWindowLimitTimeoutNs);
+        uriStringBuilder.untetheredRestingTimeoutNs(params.untetheredRestingTimeoutNs);
+
+        uriStringBuilder.mtu(params.mtuLength);
+        uriStringBuilder.termLength(params.termLength);
+        uriStringBuilder.initialTermId(params.initialTermId);
+        uriStringBuilder.termId(params.hasPosition ? params.termId : params.initialTermId);
+        uriStringBuilder.termOffset(params.hasPosition ? params.termOffset : 0);
```

Comment:
This logic is already implemented in `PublicationParams`, i.e. just assign variables directly.